internal/kafka/test/integration/kafkas_test.go

```diff
@@ -1968,3 +1997,39 @@ func TestKafka_RemovingExpiredKafkas_WithStandardInstances(t *testing.T) {
 	kafkaDeletionErr := common.WaitForNumberOfKafkaToBeGivenCount(ctx, test.TestServices.DBFactory, client, 2)
 	Expect(kafkaDeletionErr).NotTo(HaveOccurred(), "Error waiting for kafka deletion: %v", kafkaDeletionErr)
 }
+
+func getAndDeleteServiceAccounts(clientID string, env *environments.Env) {
+	var keycloakConfig *keycloak.KeycloakConfig
+	env.MustResolve(&keycloakConfig)
+	defer env.Cleanup()
+
+	kcClientKafkaSre := keycloak.NewClient(keycloakConfig, keycloakConfig.OSDClusterIDPRealm)
+	accessTokenKafkaSre, _ := kcClientKafkaSre.GetToken()
+
+	client, err := kcClientKafkaSre.GetClient(clientID, accessTokenKafkaSre)
+	if client == nil {
+		kcClient := keycloak.NewClient(keycloakConfig, keycloakConfig.KafkaRealm)
+		accessToken, _ := kcClient.GetToken()
+
+		client, _ := kcClient.GetClient(clientID, accessToken)
+		if client != nil {
+			value := *client.ID
+
+			deleteErr := kcClient.DeleteClient(value, accessToken)
+			if deleteErr != nil {
+				panic(deleteErr)
+			}
+		}
+	}
+	if err != nil {
+		fmt.Println("err")
+	}
+	if client != nil {
+		value := *client.ID
+		deleteErr := kcClientKafkaSre.DeleteClient(value, accessTokenKafkaSre)
+		if deleteErr != nil {
+			panic(deleteErr)
+		}
+	}
```

Comment:
Here we have an `if client == nil` (line 2010) where we ignore the err, then we have an `if err != nil` where we just print 'err` and finally an `if client != nil`.

I think this should be rewritten in something like:
```go
client, err := ...
if err != nil {
   // handle error. Maybe return? Or panic as in other line 2031?
}
if client != nil { // can client be nil if we don't have any error?
   // do stuff
} else {
  // handle nil client
}